lib/modules/platform/github/index.ts

```diff
@@ -144,6 +145,21 @@ export async function initPlatform({
   }
 
   await detectGhe(token);
+  /**
+   * GHE requires version >=3.10 to support fine-grained access tokens
+   * https://docs.github.com/en/enterprise-server@3.10/admin/release-notes#authentication
+   */
+  if (isGithubFineGrainedPersonalAccessToken(token) && platformConfig.isGhe) {
+    if (!platformConfig.gheVersion) {
```

Comment:
I'm unsure if this check is necessary, or if these two conditions should be combined into one with a unified error message.

I noticed a test case in `lib/modules/platform/github/index.spec.ts`: ___should support custom endpoint without version___, which adds this check when no version is detected from GHE.

However, I am unsure in which scenario GHE omits the version from the response, or if this case should prevent Renovate from proceeding.

Reply:
combine to one error, it should fail for unknown and old versions.

missing version header typically means a very old ghes.